42.md

```diff
@@ -0,0 +1,44 @@
+NIP-42
+======
+
+External Embed Sizing
+---------------
+
+`draft` `optional` `author:deanpress`
+
+An `embed-sizes` array in notes that allows relays to store information about the filesizes of external embeddable media (i.e. URLs to images and other media in posts and avatars).
+
+Clients can auto-embed small media within a configured limit, and not auto-embed larger media by default (and give the user an option to click "View Embedded Media"), to save user bandwidth and prevent abuse.
+
+Spec
+---------------
+```
+tag: embed-sizes
+values:
+ - [array of strings of filesizes in bytes, in order of urls in note content]: required
+```
+
+Example
+---------------
+```json
+{
+    "pubkey": "<pub-key>",
+    "created_at": 1000000000,
+    "kind": 1,
+    "tags": [
+      ["embed-sizes", ["500000", "1500000", "0"]]
```

Comment:
We only save the embed size, but is that all we want to save? Might be better to embed the hash of the content as well.

Reply:
Agree we probably need more. But how does a client utilize the hash of an external embed without having downloaded the file first?